<html>
<head>
<title>How to Normalize Single Cell</title>
<link href="../Styles/Style.css" type="text/css" rel="stylesheet"/>
</head>
<body>
<h1 class="translated">如何使单个细胞正常化</h1>
<blockquote>原文：<a href="https://towardsdatascience.com/how-to-normalize-single-cell-a438281ea654?source=collection_archive---------8-----------------------#2019-08-23">https://towardsdatascience.com/how-to-normalize-single-cell-a438281ea654?source=collection_archive---------8-----------------------#2019-08-23</a></blockquote><div><div class="fc ih ii ij ik il"/><div class="im in io ip iq"><h2 id="a9ea" class="ir is it bd b dl iu iv iw ix iy iz dk ja translated" aria-label="kicker paragraph"><a class="ae ep" href="https://towardsdatascience.com/tagged/stats-ml-life-sciences" rel="noopener" target="_blank">生命科学的数理统计和机器学习</a></h2><div class=""/><div class=""><h2 id="209c" class="pw-subtitle-paragraph jz jc it bd b ka kb kc kd ke kf kg kh ki kj kk kl km kn ko kp kq dk translated">单细胞基因组学标准化策略的比较</h2></div><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi kr"><img src="../Images/59cd582c9c3b08f04b800fc663ddddef.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*fbyUfWI8WFVSDaqV8tn3HQ.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">From C. Hafemeister et al. 2019, <a class="ae lh" href="https://www.biorxiv.org/content/biorxiv/early/2019/03/18/576827.full.pdf" rel="noopener ugc nofollow" target="_blank">image source</a></figcaption></figure><p id="a8ac" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">这是专栏<a class="ae lh" href="https://towardsdatascience.com/tagged/stats-ml-life-sciences?source=post_page---------------------------" rel="noopener" target="_blank"> <strong class="lk jd">生命科学的数理统计和机器学习</strong> </a>的第九篇文章，我试图在这里涵盖生物信息学、生物医学、遗传学、进化科学等领域常用的分析技术。今天，我们将通过比较最流行的<strong class="lk jd">归一化</strong>算法，来讨论在激动人心的<a class="ae lh" href="https://en.wikipedia.org/wiki/Single_cell_sequencing" rel="noopener ugc nofollow" target="_blank">单细胞基因组学</a>领域中<strong class="lk jd">消除</strong> <strong class="lk jd">技术变异</strong>的挑战。</p><h1 id="cacc" class="me mf it bd mg mh mi mj mk ml mm mn mo ki mp kj mq kl mr km ms ko mt kp mu mv bi translated">从批量标准化到单细胞标准化</h1><p id="3e5c" class="pw-post-body-paragraph li lj it lk b ll mw kd ln lo mx kg lq lr my lt lu lv mz lx ly lz na mb mc md im bi translated">人们普遍认为，<a class="ae lh" href="https://en.wikipedia.org/wiki/Single-cell_transcriptomics" rel="noopener ugc nofollow" target="_blank">单细胞 RNA 测序(<strong class="lk jd"> scRNAseq </strong> ) </a>实验中的<a class="ae lh" rel="noopener" target="_blank" href="/pitfalls-of-data-normalization-bf05d65f1f4c">大规模细胞间基因表达变异</a>可能在很大程度上是<strong class="lk jd">技术性的</strong>，从而混淆了与细胞类型相关的<strong class="lk jd">生物学</strong> <strong class="lk jd">变异</strong>。技术变异的来源各不相同，从<a class="ae lh" href="https://en.wikipedia.org/wiki/Polymerase_chain_reaction" rel="noopener ugc nofollow" target="_blank"> PCR 扩增</a>偏差到<a class="ae lh" href="https://en.wikipedia.org/wiki/Reverse_transcriptase" rel="noopener ugc nofollow" target="_blank">逆转录(RT) </a>效率以及制备和测序步骤中分子取样的随机性。因此，必须应用跨细胞的标准化，以消除技术变异并保持生物学变异。然而，常见的<strong class="lk jd">批量 RNAseq </strong>方法，如<a class="ae lh" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd"> TMM </strong> </a>和<a class="ae lh" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-10-r106" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd"> DESeq </strong> </a>都是基于构建每个单元的<strong class="lk jd">大小因子</strong>，这些因子代表<strong class="lk jd">比率</strong>，其中<strong class="lk jd"> </strong>每个单元都由一个<strong class="lk jd">参考单元</strong>进行归一化，该参考单元是通过对所有其他单元进行某种<strong class="lk jd">平均</strong>而构建的。</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nb"><img src="../Images/0e51d4848d83f6c13cc7f0ccc06a12a4.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*1IPZ4GonO4GuzIWLRpF8JQ.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Bulk normalization (TMM, DESeq) builds a reference sample, from Mar Gonzalez-Porta <a class="ae lh" href="https://github.com/mgonzalezporta" rel="noopener ugc nofollow" target="_blank">teaching material</a></figcaption></figure><p id="8917" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">scRNAseq 数据的一个特点是，与大量 RNAseq 相比，它们包含大量的<strong class="lk jd">随机零</strong>，这是由于每个细胞的 RNA 量低和 RNA 捕获效率不理想(<strong class="lk jd">丢失</strong>)。因此，TMM 和 DESeq <strong class="lk jd">尺寸因子</strong>可能变得不可靠<strong class="lk jd">膨胀或等于零</strong>。因此，开发了新的单细胞特异性标准化方法，解决了 scRNAseq 数据中大量零的问题。</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nc"><img src="../Images/fc9ce1bdc556c232e64c95d8e7562922.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*_W9iMjQPi2eECMif3x9szw.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Genes expressed in one cell are missing in another cell due to <a class="ae lh" href="https://www.nature.com/articles/nmeth.2967" rel="noopener ugc nofollow" target="_blank">dropout</a> (left), zero inflated scRNAseq data (right)</figcaption></figure><p id="70a0" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">在<a class="ae lh" href="https://www.ebi.ac.uk/research/marioni" rel="noopener ugc nofollow" target="_blank"> Marioni 的实验室</a>中，用<a class="ae lh" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd">反卷积归一化</strong> </a>方法对归一化过程中的随机零点进行了一次非常优雅的尝试。该算法背后的思想是通过多个<strong class="lk jd">池</strong>单元来表示所有单元，对于每个池，来自多个单元的表达式值被求和，这导致<strong class="lk jd">更少的零</strong>，并且通过所有单元的<strong class="lk jd">平均表达式</strong>来缩放。以这种方式构建的每池尺寸因子<strong class="lk jd">可以在线性方程组</strong>中<strong class="lk jd">解卷积</strong> <strong class="lk jd">，产生每单元尺寸因子</strong>的<strong class="lk jd">。</strong></p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nd"><img src="../Images/19e95226947248e4fb1e5c49e87d3704.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*48Q4LaLz2djSJCpoeeqjNQ.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Pooling across cell, from <a class="ae lh" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7" rel="noopener ugc nofollow" target="_blank">A. Lun et al., Genome Biology 2017</a></figcaption></figure><p id="7fb3" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">这种“整体大小因子方法”的缺点是，不同组的基因不应该用每个细胞大小因子的相同常数来标准化，而是应该针对每组基因<strong class="lk jd"/>例如高、中、低表达的基因<strong class="lk jd"> </strong>进行单独调整。这在<a class="ae lh" href="https://www.nature.com/articles/nmeth.4263" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd"> SCnorm </strong> </a>和更近的<a class="ae lh" href="https://www.biorxiv.org/content/10.1101/576827v2.full" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd">皮尔逊残差</strong> </a>归一化算法中被考虑在内。两者都是基于回归出不同基因组的测序深度偏差。</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi ne"><img src="../Images/f03a24aba4e4f42b4162a4b5dc517e91.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*zOzCmIPNBnm2XXxOPkGSeQ.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">SCnorm builds per cell per group of genes size factors, from <a class="ae lh" href="https://www.nature.com/articles/nmeth.4263" rel="noopener ugc nofollow" target="_blank">Bacher et al., Nature Methods 2017</a></figcaption></figure><p id="c90f" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">下面我们将使用来自<a class="ae lh" href="https://www.nature.com/articles/ni.3368" rel="noopener ugc nofollow" target="_blank">的<strong class="lk jd">先天淋巴细胞(ILC) </strong> scRNAseq 数据来比较不同的流行标准化策略。bjrklund 等人，<em class="nf">自然免疫学</em> <strong class="lk jd"> 17 </strong>，第 451–460 页(2016) </a></p><h1 id="eed1" class="me mf it bd mg mh mi mj mk ml mm mn mo ki mp kj mq kl mr km ms ko mt kp mu mv bi translated">比较 scRNAseq 归一化方法</h1><p id="dcb5" class="pw-post-body-paragraph li lj it lk b ll mw kd ln lo mx kg lq lr my lt lu lv mz lx ly lz na mb mc md im bi translated">scRNAseq 分析工作流程相当复杂，在这里我将专注于结果，但鼓励您查看我的<a class="ae lh" href="https://github.com/NikolayOskolkov/NormalizeSingleCell" rel="noopener ugc nofollow" target="_blank"> github </a>中的完整笔记本。首先，ILC scRNAseq 数据集是用<a class="ae lh" href="https://www.ncbi.nlm.nih.gov/pubmed/24385147" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd"> SmartSeq2 </strong> </a>全长转录物测序技术产生的，并且包括非常方便的<strong class="lk jd">尖峰信号</strong>(跨细胞恒定浓度的对照转录物)用于评估技术<strong class="lk jd">背景噪音</strong>。通过将<a class="ae lh" href="https://en.wikipedia.org/wiki/Coefficient_of_variation" rel="noopener ugc nofollow" target="_blank">变异系数</a> (CV=sd/mean)绘制为每种标准化方法的平均基因表达的函数，这些峰值可用于检测变异高于噪声水平的一组基因:</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi ng"><img src="../Images/603803ae1e0d0295f9545a4d7009552d.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*unIZ1PeB92dXJ77awtNviw.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Detecting genes with variation above technological noise using spike-ins</figcaption></figure><p id="0641" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">在这些图上:一个点是一个基因，红色曲线代表峰值的变异，峰值曲线上方的<strong class="lk jd">基因</strong>展示了技术变异上方的变异(<strong class="lk jd">高度变异基因</strong>)。可以观察到，可变基因的数量根据标准化方法而变化。<strong class="lk jd"> RPKM </strong> <strong class="lk jd">归一化显示了</strong> <strong class="lk jd">最低数量的可变基因</strong>，这可能是由于在 RPKM 中用于归一化的短长度的刺突。接下来，我们将检查<strong class="lk jd">主成分分析(PCA)图</strong>如何根据 scRNAseq 归一化策略而变化:</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nh"><img src="../Images/d753d1bca267108b88020a93123133d8.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*QzKrgk-XiXwUAMiwjW7-7A.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">PCA plots for different scRNAseq normalization strategies</figcaption></figure><p id="36d9" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">很难看出标准化方法之间的任何差异，PCA 图看起来<strong class="lk jd">几乎相同</strong>，表明来自<strong class="lk jd"> 4 个 ILC 簇</strong>的细胞严重重叠。此外，如预期的那样，我们可以使用<a class="ae lh" href="https://en.wikipedia.org/wiki/T-distributed_stochastic_neighbor_embedding" rel="noopener ugc nofollow" target="_blank"> <strong class="lk jd"> tSNE </strong> </a> <strong class="lk jd">图</strong>获得 ILC 细胞的更明显的聚类:</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi ni"><img src="../Images/1c457acf3e0e7329b368081fd65287b7.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*Qrfk4Kzs9Z2KynDYtgAxwQ.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">tSNE plots for different scRNAseq normalization strategies</figcaption></figure><p id="8ac6" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">然而，归一化的方法似乎并没有改变 4 个 ILC 簇，不管采用什么归一化方法，它们都非常清晰可见。最后，我们可以计算不同归一化策略的大小因子，并检查它们的相关性:</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nj"><img src="../Images/f6db1c21382a21e2bcca30d360841783.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*wSr6LAoaBJjakFzlaeKHbA.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Comparing size factors across different normalization strategies</figcaption></figure><p id="4de6" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">这里为了简单起见，我对不同基因组的 SCnorm 大小因子进行了平均。令人惊讶的是，<strong class="lk jd">大小因子在不同的归一化方法中几乎是不可区分的</strong>，甚至包括批量 RNAseq 算法 TMM 和 DEseq。然而，对于其他 scRNAseq 数据集，这可能不如 ILC 看起来好。</p><p id="84c7" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">看着上面的图，感觉上对 scRNAseq 的标准化的重要性被稍微高估了，因为至少对于 SmartSeq2 数据集，性能上的<strong class="lk jd">差异是可以忽略的</strong>。如果细胞群体足够不同，即如果数据中有信号，选择哪种 scRNAseq 归一化方法并不重要。此外，随着基于 UMI 的测序方案的广泛使用，对复杂标准化算法的需求甚至变得<strong class="lk jd">不那么明显</strong>，因为这些方案原则上能够消除扩增和测序深度偏差，因为多个读数被压缩到单个 UMI 计数中。</p><figure class="ks kt ku kv gt kw gh gi paragraph-image"><div role="button" tabindex="0" class="kx ky di kz bf la"><div class="gh gi nk"><img src="../Images/976708bdbb4e6c9745ad5a47ba64c383.png" data-original-src="https://miro.medium.com/v2/resize:fit:1400/format:webp/1*2Sddd_54hhgsjU1P6BtJTw.png"/></div></div><figcaption class="ld le gj gh gi lf lg bd b be z dk">Major sources of technical variation in scRNAseq that can be improved by UMI-based protocols, <a class="ae lh" href="https://www.nature.com/articles/nmeth.4292" rel="noopener ugc nofollow" target="_blank">images source</a></figcaption></figure><h1 id="205a" class="me mf it bd mg mh mi mj mk ml mm mn mo ki mp kj mq kl mr km ms ko mt kp mu mv bi translated">摘要</h1><p id="3084" class="pw-post-body-paragraph li lj it lk b ll mw kd ln lo mx kg lq lr my lt lu lv mz lx ly lz na mb mc md im bi translated">在这篇文章中，我们了解到由于 scRNAseq 数据集中大量的随机零，基于<strong class="lk jd">比率的</strong> <strong class="lk jd">传统批量 RNAseq </strong>归一化算法<strong class="lk jd">不适用</strong>。去卷积和 SCnorm 代表了优雅的特定于 scRNAseq 的归一化方法。然而，scRNAseq 标准化方法学<strong class="lk jd">的比较并未显示其性能</strong><strong class="lk jd"/>的显著差异，随着基于 UMI 的 scRNAseq 方案的进展，这可能变得更加不显著。</p><p id="1306" class="pw-post-body-paragraph li lj it lk b ll lm kd ln lo lp kg lq lr ls lt lu lv lw lx ly lz ma mb mc md im bi translated">在下面的评论中让我知道生命科学中的哪些分析对你来说似乎特别神秘，我会在这个专栏中尝试回答它们。在我的<a class="ae lh" href="https://github.com/NikolayOskolkov/NormalizeSingleCell" rel="noopener ugc nofollow" target="_blank"> github </a>上查看帖子中的代码。在媒体<a class="nl nm ep" href="https://medium.com/u/8570b484f56c?source=post_page-----a438281ea654--------------------------------" rel="noopener" target="_blank">关注我，在 Twitter @NikolayOskolkov 关注我，在 Linkedin </a>关注我。下期我们将覆盖<strong class="lk jd">如何批量修正单细胞数据</strong>，敬请关注。</p></div></div>    
</body>
</html>